llvm/include/llvm/IR/IntrinsicsNVVM.td

```diff
@@ -5381,4 +5381,50 @@ def int_nvvm_st_bulk_shared_cta : DefaultAttrsIntrinsic<[],
   [IntrArgMemOnly, IntrWriteMem,
     WriteOnly<ArgIndex<0>>, NoCapture<ArgIndex<0>>, ImmArg<ArgIndex<2>>]>;
 
+//
+// Cluster launch control
+//
+
+// clusterlaunchcontrol.try_cancel
+
+def int_nvvm_clusterlaunchcontrol_try_cancel_async
+    : Intrinsic<[], [llvm_ptr_ty, llvm_ptr_ty],
+                [IntrHasSideEffects, IntrArgMemOnly, NoCapture<ArgIndex<0>>, NoCapture<ArgIndex<1>>],
```

Comment:
Instruction documentation says "asynchronously writes an opaque response to shared memory". The way I read it, it means that intrinsic does effectively capture the pointer. I.e. it will continue to be in use after the instruction has returned.

Reply:
Yes, after the instruction returns, some time later, there will be a write to the shared memory pointer, and an arrival to the barrier (both).

Reply:
Removed `NoCapture` property as instruction captures both addr and mbar operand and writes to it later